sw/device/lib/dif/dif_rv_core_ibex.c

```diff
@@ -0,0 +1,176 @@
+                .ibus_addr_en = RV_CORE_IBEX_IBUS_ADDR_EN_0_REG_OFFSET,
+                .ibus_reg_en = RV_CORE_IBEX_IBUS_REGWEN_0_REG_OFFSET,
+                .dbus_addr_maching =
+                    RV_CORE_IBEX_DBUS_ADDR_MATCHING_0_REG_OFFSET,
+                .dbus_addr_remap = RV_CORE_IBEX_DBUS_REMAP_ADDR_0_REG_OFFSET,
+                .dbus_addr_en = RV_CORE_IBEX_DBUS_ADDR_EN_0_REG_OFFSET,
+                .dbus_reg_en = RV_CORE_IBEX_DBUS_REGWEN_0_REG_OFFSET,
+            },
+        [kDifRvCoreIbexAddrTranslationSlot_1] =
+            {
+                .ibus_addr_maching =
+                    RV_CORE_IBEX_IBUS_ADDR_MATCHING_1_REG_OFFSET,
+                .ibus_addr_remap = RV_CORE_IBEX_IBUS_REMAP_ADDR_1_REG_OFFSET,
+                .ibus_addr_en = RV_CORE_IBEX_IBUS_ADDR_EN_1_REG_OFFSET,
+                .ibus_reg_en = RV_CORE_IBEX_IBUS_REGWEN_1_REG_OFFSET,
+                .dbus_addr_maching =
+                    RV_CORE_IBEX_DBUS_ADDR_MATCHING_1_REG_OFFSET,
+                .dbus_addr_remap = RV_CORE_IBEX_DBUS_REMAP_ADDR_1_REG_OFFSET,
+                .dbus_addr_en = RV_CORE_IBEX_DBUS_ADDR_EN_1_REG_OFFSET,
+                .dbus_reg_en = RV_CORE_IBEX_DBUS_REGWEN_1_REG_OFFSET,
+            },
+};
+
+/**
+ * Get the list of register from the look up table.
+ *
+ * @param id Slot id.
+ * @return struct ibex_addr_translation_regs.
+ */
+static inline struct ibex_addr_translation_regs get_regs_from_slot(
+    dif_rv_core_ibex_addr_translation_slot_t id) {
+  return kRegs_map[id];
+}
+
+/**
+ * Convert the region address and size into a natural power of two address.
+ *
+ * @param addr Region start address.
+ * @param size region size.
+ * @return Napot address
+ */
+static inline uint32_t to_napot(uint32_t addr, size_t size) {
+  return addr | (size - 1) >> 1;
+}
+
+/**
+ * Split a natural power of two addres into a start address and size.
+ *
+ * @param napot Address formated in NAPOT.
+ * @param size  Pointer to receive the region size.
+ * @return The region start address.
+ */
+static inline uint32_t from_napot(uint32_t napot, size_t *size) {
+  for (size_t i = 0; i < sizeof(uint32_t) * 8; ++i) {
+    uint32_t ref = (0x01 << i) - 1;
+    if ((napot & ref >> 1) == ref >> 1) {
+      *size = 0x01 << i;
+    }
+  }
+  return napot & ~((*size - 1) >> 1);
+}
+
+/**
+ * Check whether the size is power of two algned.
+ *
+ * @param size Size to be verified.
+ * @return true I the size is power of two, otherwise false.
+ */
+static inline bool is_power_of_two(size_t size) {
+  return size && !(size & (size - 1));
+}
+
+dif_result_t dif_rv_core_ibex_configure_addr_translation(
+    const dif_rv_core_ibex_t *rv_core_ibex,
+    dif_rv_core_ibex_addr_translation_slot_t slot,
+    dif_rv_core_ibex_addr_translation_region_t region) {
+  if (rv_core_ibex == NULL || slot >= kDifRvCoreIbexAddrTranslationSlotCount) {
+    return kDifBadArg;
+  }
+
+  if (!(is_power_of_two(region.dbus.size) &&
```

Comment:
Maybe apply de Morgan's law to write `!is_power_of_two || !is_power_of_two` to avoid expression nesting.

Reply:
Done